great_expectations/exceptions/exceptions.py

```diff
@@ -410,3 +410,9 @@ class GeCloudError(GreatExpectationsError):
     """
 
     pass
+
+
+class TestDatabaseConnectionError(GreatExpectationsError):
```

Comment:
@anthonyburdi Does this need to be specific to "Test" or can this be a "Generaal Database Connection Error"?  Thanks.

Reply:
Great question. My rationale is that I added the `Test` prefix so that in case we log our error types as usage events in the future (to help proactively fix user issues) we might want to separate connection errors where users are trying to connect to dbs with connection errors of our own CI pipeline.

Reply:
@anthonyburdi I feel that this distinction should be accomplished differently, if possible.

Reply:
Upon reflection - this distinction is unnecessary since events logged in our CI pipeline are already tracked differently. I'll change this to `DatabaseConnectionError`. Thank you for the feedback and discussion!

Reply:
This has now been updated @alexsherstinsky 🙇